### PR TITLE
Objectlist

### PR DESCRIPTION
**Pull request by bouda1**
Again some cleanup and new things to work on notification.

### CODE REVIEW

File: src/host.cc [hunk 1]
```diff
@@ -3358,9 +3336,8 @@ int host::process_check_result_3x(enum host::host_state new_state,
                                  int use_cached_result,
                                  unsigned long check_timestamp_horizon) {
   com::centreon::engine::host* master_host = NULL;
-  com::centreon::engine::host* temp_host = NULL;
-  objectlist* check_hostlist = NULL;
-  objectlist* hostlist_item = NULL;
+  host* temp_host;
+  std::list<host*> check_hostlist;
```
Comment (SylvestreG) [inline src/host.cc]: you can perhaps switch to std::list<shared_ptr<host>> here

File: src/host.cc [hunk 2]
```diff
@@ -3358,9 +3336,8 @@ int host::process_check_result_3x(enum host::host_state new_state,
                                  int use_cached_result,
                                  unsigned long check_timestamp_horizon) {
   com::centreon::engine::host* master_host = NULL;
-  com::centreon::engine::host* temp_host = NULL;
-  objectlist* check_hostlist = NULL;
-  objectlist* hostlist_item = NULL;
+  host* temp_host;
```
Comment (SylvestreG) [inline src/host.cc]: i think you can remove this variable and use directly the iterator.

File: src/service.cc [hunk 3]
```diff
@@ -1058,8 +1053,7 @@ int service::handle_async_check_result(check_result* queued_check_result) {
   time_t current_time = 0L;
   int state_was_logged = false;
   std::string old_plugin_output;
-  objectlist* check_servicelist = nullptr;
-  objectlist* servicelist_item = nullptr;
+  std::list<service*> check_servicelist;
```
Comment (SylvestreG) [inline src/service.cc]: if it is possible you can perhaps switch to std::list<shared_ptr<service>>

File: src/host.cc (hunk 1 above)
Comment (bouda1) [reply src/host.cc]: This list is just some kind of reminder it is removed at the end of the function. And for several hosts, I don't have the shared_ptr container, only the host itself.

File: src/service.cc (hunk 3 above)
Comment (bouda1) [reply src/service.cc]: The response as for hosts

**Pull request merged**